pkg/ingester/lookupplan/planner.go

```diff
@@ -37,6 +37,11 @@ func NewCostBasedPlanner(metrics Metrics, statistics Statistics) *CostBasedPlann
 }
 
 func (p CostBasedPlanner) PlanIndexLookup(ctx context.Context, inPlan index.LookupPlan, _, _ int64) (_ index.LookupPlan, retErr error) {
+	// If planning is disabled, return the input plan directly
+	if isDisabledPlanning(ctx) {
```

Comment:
Shorter and easier to understand (imo)

```suggestion
	if planningDisabled(ctx) {
```